mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoMethodGenerator.java

```diff
@@ -234,4 +244,30 @@ protected boolean isFromClassFile() {
     TypeElement enclosingElement = (TypeElement) methodElement.getEnclosingElement();
     return Reflection.loadClass(null, enclosingElement.getQualifiedName().toString()) != null;
   }
+
+  /**
+   * Common pattern for CRUD methods that build a bound statement, execute it and convert the result
+   * into a target type.
+   *
+   * @param createStatementBlock the code that creates the statement. It must store it into a
+   *     variable named "boundStatement".
+   */
+  protected Optional<MethodSpec> crudMethod(
+      CodeBlock.Builder createStatementBlock, DaoReturnType returnType, String helperFieldName) {
+
+    MethodSpec.Builder method = GeneratedCodePatterns.override(methodElement, typeParameters);
+    TypeName returnTypeName = null;
+    if (returnType.getKind() == CUSTOM) {
```

Comment:
it's a bit odd the the `producer` variable declaration is generated here, but only accessed by code generated inside `addExecuteStatement`. Instead of special-casing CUSTOM here, can't you generate the `producer` variable delcaration inside `DefaultDaoReturnTypeKind.CUSTOM.addExecuteStatement()` ?

Reply:
The output of `addExecuteStatement()` is enclosed in the `try` block. We also need the producer in the `catch` block to wrap errors.